client/blocks/plan-purchase-features/index.jsx

```diff
@@ -0,0 +1,243 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PLAN_FREE,
+	PLAN_PERSONAL,
+	PLAN_PREMIUM,
+	PLAN_BUSINESS,
+	PLAN_JETPACK_FREE,
+	PLAN_JETPACK_PREMIUM,
+	PLAN_JETPACK_BUSINESS,
+	PLAN_JETPACK_PREMIUM_MONTHLY,
+	PLAN_JETPACK_BUSINESS_MONTHLY
+} from 'lib/plans/constants';
+import {
+	FindNewThemeFeature,
+	AdvertisingRemovedFeature,
+	GoogleVouchersFeature,
+	CustomizeThemeFeature,
+	VideoAudioPostsFeature,
+	MonetizeSiteFeature,
+	CustomDomainFeature,
+	GoogleAnalyticsStatsFeature,
+	JetpackFeatures,
+	HapinessSupportFeature,
+	CurrentPlanHeaderFeature
+} from './features-list';
+import { getPlansBySite } from 'state/sites/plans/selectors';
+import { getSelectedSite } from 'state/ui/selectors';
+import {
+	isPremium,
+	isBusiness
+} from 'lib/products-values';
+import { isEnabled } from 'config';
+import paths from 'lib/paths';
+import { isWordadsInstantActivationEligible } from 'lib/ads/utils';
+import userFactory from 'lib/user';
+import analytics from 'lib/analytics';
+import utils from 'lib/site/utils';
+
+class PlanPurchaseFeatures extends Component {
+	static propTypes = {
+		plan: PropTypes
+			.oneOf( [ PLAN_FREE, PLAN_PERSONAL, PLAN_PREMIUM, PLAN_BUSINESS ] )
+			.isRequired
+	};
+
+	isCustomizeEnabled() {
+		return isEnabled( 'manage/customize' );
+	}
+
+	getCustomizeLink() {
+		const { selectedSite } = this.props;
+
+		const adminUrl = selectedSite.URL + '/wp-admin/',
+			customizerInAdmin = adminUrl + 'customize.php?return=' + encodeURIComponent( window.location.href );
+
+		return this.isCustomizeEnabled() ? '/customize/' + selectedSite.slug : customizerInAdmin;
+	}
+
+	getBusinessFeatures() {
+		const {
+			selectedSite,
+			sitePlans,
+			translate
+		} = this.props;
+
+		const plan = find( sitePlans.data, isBusiness ),
+			hasLoadedFromServer = sitePlans.hasLoadedFromServer;
+
+		return [
+			<HapinessSupportFeature
+				selectedSite={ selectedSite }
+				key="hapinessSupportFeature"
+			/>,
+			<CurrentPlanHeaderFeature
+				selectedSite={ selectedSite }
+				key="currentPlanHeaderFeature"
+				hasLoadedFromServer={ hasLoadedFromServer }
+				title={ translate( 'Your site is on a Business plan' ) }
+				tagLine={ translate( 'Learn more about everything included with Business and take advantage of' +
+					' its professional features.' ) }
+			/>,
+			<CustomDomainFeature
+				selectedSite={ selectedSite }
+				hasDomainCredit={ plan && plan.hasDomainCredit }
+				key="customDomainFeature"
+			/>,
+			<FindNewThemeFeature
+				selectedSite={ selectedSite }
+				key="findNewThemeFeature"
+			/>,
+			<GoogleAnalyticsStatsFeature
```

Comment:
Alright, I see your points (both @artpi's and @rralian's). However, I would need to pass both `selectedSite` and `sitePlans` to all the feature components because some of them need both of these. That doesn't seem to be an issue so I'm going for this.